### PR TITLE
Dbraunst part1 - submission

### PR DESCRIPTION
**Pull request by dbraunst**
Please review submission for p7 part 1

### CODE REVIEW

File: Utility.h [hunk 1]
```diff
@@ -5,13 +5,15 @@
 #include "Item.h"
 
 
-std::vector<std::unique_ptr<Item>> makeHelpfulItems(int num);
+std::vector<std::unique_ptr<Item>> makeHelpfulItems();
```
Comment (matkatmusic) [inline Utility.h]: Why did you change this?  
You could have called `makeHelpfulItems( rand() % 3 );`

File: Utility.cpp [hunk 2]
```diff
@@ -3,10 +3,19 @@
...
 
+#include "Paladin.h"
+#include "Dwarf.h"
+#include "Dragon.h"
+#include "DragonSlayer.h"
 
-std::vector<std::unique_ptr<Item>> makeHelpfulItems(int num)
+std::vector<std::unique_ptr<Item>> makeHelpfulItems()
 {
+    int num = rand() % 3;
```
Comment (matkatmusic) [inline Utility.cpp]: Why did you change this?
You could have called makeHelpfulItems( rand() % 3 );

File: Utility.cpp [hunk 3]
```diff
@@ -18,8 +27,10 @@ std::vector<std::unique_ptr<Item>> makeHelpfulItems(int num)
     return items;
 }
 
-std::vector<std::unique_ptr<Item>> makeDefensiveItems(int num)
+std::vector<std::unique_ptr<Item>> makeDefensiveItems()
 {
+    int num = rand() % 3;
```
Comment (matkatmusic) [inline Utility.cpp]: Same question as above.

File: Utility.cpp [hunk 4]
```diff
@@ -31,6 +42,21 @@ std::vector<std::unique_ptr<Item>> makeDefensiveItems(int num)
     return items;
 }
 
+std::vector<std::unique_ptr<Item>> makeAttackItems()
+{
+    int num = rand() % 3;
```
Comment (matkatmusic) [inline Utility.cpp]: Why not conform to the previous interface pattern: `make____Item(int num)` ?

**Comment by dbraunst** [comment]
I thought when requesting the specific range of items to be created you also wanted me to remove the in-call functionality and to set it in the function itself

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dbraunst** [comment]
I can change it back if you’d like

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
